sdk/core/azure-core-test/src/main/java/com/azure/core/test/http/PlaybackClient.java

```diff
@@ -121,7 +124,20 @@ public HttpClient port(int i) {
                 }
             }
 
-            bytes = rawBody.getBytes(StandardCharsets.UTF_8);
+            String contentType = networkCallRecord.response().get("Content-Type");
+
+            // octet-stream's are written to disk using Arrays.toString() which creates an output such as "[12, -1]".
+            if (contentType != null && contentType.contains("octet-stream")) {
```

Comment:
"application/octet-stream" mime-type is well known. I'd use `.equalsIgnoreCase()` because they are also case-insensitive.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types

Reply:
in the RFC, headers are case insensitive. It is by convention that you see "Content-Type" rather than "content-type". But they amount to the same thing.

Reply:
Playback client is not validator. Is it better to ignore the case for all the hard-code string checking?

Reply:
RecordNetworkCallPolicy encounters the same issues. I added a comment about that.

Reply:
While RFC headers are case-insensitive the data structure, `Map`, we use to store this information in playback isn't. This is another cleanup we should do to either use `HttpHeaders` or a case-insensitive `Map` implementation.

Reply:
SO has revealed that there is a way to have case insensitive key comparison. One thing that .NET provides over HashMap.
https://stackoverflow.com/a/22336599

Reply:
Updated to the case ignores check.